From 7fd0d2577a2c39ad5c6b823a65909aba737e419a Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sat, 18 Jan 2020 14:49:16 +0100
Subject: [PATCH] Add missing cstddef includes

Fixes build errors seen with gcc 5.5.0:

xbmc/pictures/SlideShowPicture.cpp:903:109: error: 'offsetof' was not declared in this scope
   glVertexAttribPointer(posLoc, 3, GL_FLOAT, 0, sizeof(PackedVertex), BUFFER_OFFSET(offsetof(PackedVertex, x)));

xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.cpp:187:109: error: 'offsetof' was not declared in this scope
   glVertexAttribPointer(posLoc, 3, GL_FLOAT, 0, sizeof(PackedVertex), BUFFER_OFFSET(offsetof(PackedVertex, x)));
---
 .../rendering/VideoRenderers/RPRendererGuiTexture.cpp           | 2 ++
 xbmc/pictures/SlideShowPicture.cpp                              | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.cpp b/xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.cpp
index 01f30710daee..80bf61a468c5 100644
--- a/xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.cpp
+++ b/xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.cpp
@@ -17,6 +17,8 @@
 using namespace DirectX;
 #endif
 
+#include <cstddef>
+
 using namespace KODI;
 using namespace RETRO;
 
diff --git a/xbmc/pictures/SlideShowPicture.cpp b/xbmc/pictures/SlideShowPicture.cpp
index b6516ce9b8c6..fe4b3057fc58 100644
--- a/xbmc/pictures/SlideShowPicture.cpp
+++ b/xbmc/pictures/SlideShowPicture.cpp
@@ -32,6 +32,8 @@ using namespace DirectX;
 using namespace Microsoft::WRL;
 #endif
 
+#include <cstddef>
+
 #define IMMEDIATE_TRANSITION_TIME          20
 
 #define PICTURE_MOVE_AMOUNT              0.02f
